core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -572,6 +572,7 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
   }
 
   def shutdown(): Unit = {
+    metrics.removeSensor(delayQueueSensor.name())
```

Comment:
we should clean up the metric at the very end of the shutdown call (inside a finally). This is because the metric may still get values while the thread hasn't completed shutdown.

Reply:
I'll adjust. Thanks